cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/index.js

```diff
@@ -35,31 +37,49 @@ const browserMap = {
 };
 
 export default class DataPrepBrowser extends Component {
-  constructor(props) {
-    super(props);
-    let store = DataPrepBrowserStore.getState();
-    this.state = {
-      activeBrowser: store.activeBrowser
-    };
-  }
-  componentWillMount() {
-    DataPrepBrowserStore.subscribe(() => {
+  state = {
+    activeBrowser: DataPrepBrowserStore.getState().activeBrowser
+  };
+
+  componentDidMount() {
+    this.storeSubscription = DataPrepBrowserStore.subscribe(() => {
       let {activeBrowser} = DataPrepBrowserStore.getState();
       if (activeBrowser.name && this.state.activeBrowser.name !== activeBrowser.name) {
         this.setState({
           activeBrowser
         });
       }
     });
+    if (typeof this.props.setActiveBrowser === 'function') {
+      this.props.setActiveBrowser();
+    }
+  }
+
+  componentDidUpdate() {
+    if (typeof this.props.setActiveBrowser === 'function') {
+      this.props.setActiveBrowser();
+    }
   }
+
+  componentWillUnmount() {
+    if (typeof this.storeSubscription === 'function') {
+      this.storeSubscription();
+    }
+  }
+
   render() {
     let activeBrowser = this.state.activeBrowser.name.toLowerCase();
     if (browserMap.hasOwnProperty(activeBrowser)) {
       let Tag = browserMap[activeBrowser];
       return (
-        <Tag
-          {...this.props}
-        />
+        <Provider store={DataPrepBrowserStore}>
+          <React.Fragment>
```

Comment:
is this Fragment required?

Reply:
Yeah, if I don't have the fragment I get this error: "Warning: Failed prop type: Invalid prop `children` of type `array` supplied to `Provider`, expected a single ReactElement."